src/miner_util.erl

```diff
@@ -6,17 +6,37 @@
 -module(miner_util).
 
 -export([
+         list_count/1,
+         list_count_and_sort/1,
          index_of/2,
          h3_index/3,
          median/1,
          mark/2,
          metadata_fun/0
         ]).
 
-%%--------------------------------------------------------------------
-%% @doc
+%%-----------------------------------------------------------------------------
+%% @doc Count the number of occurrences of each element in the list.
+%% @end
+%%-----------------------------------------------------------------------------
+-spec list_count([A]) ->
+    #{A => pos_integer()}.
+list_count(Xs) ->
+    lists:foldl(
+        fun(X, Counts) -> maps:update_with(X, fun(C) -> C + 1 end, 1, Counts) end,
+        #{},
+        Xs
+    ).
+
+%%-----------------------------------------------------------------------------
+%% @doc `list_count` then sort from largest-first (head) to smallest-last.
 %% @end
-%%--------------------------------------------------------------------
+%%-----------------------------------------------------------------------------
+-spec list_count_and_sort([A]) ->
+    [{A, pos_integer()}].
+list_count_and_sort(Xs) ->
+    lists:sort(fun({_, C1}, {_, C2}) -> C1 > C2 end, maps:to_list(list_count(Xs))).
```

Comment:
this can also be `lists:keysort(2, ...)`

Reply:
True, but this way the sort order is not obscured!